Kanchanak update version

Kanchanakkh opened the pull request.

Kanchanakkh committed:
option: change layout of Navbar for landingpage. 

Make one navbar to render for cases when user logins and when user is not logged in

Kanchanakkh committed:
option: change layout of Navbar for landingpage. 

 Make one navbar to render for cases when user logins and when user is not logged in

csarahc commented:
After pulling from this branch,I found the layout of Navbar to be buggy on my end, when clicking on a button that links to another page, it does not link forward to other pages and refreshes to the previous page. Node module dependency errors also occured

Kanchanakkh committed:
modified category schema for the backend

For this new category schema, I include the minimum length and maximum length of the name. Note that the slug of the category is unique and can't be duplicate. When you add into the database, it will notify an error saying that the string must be unique

Kanchanakkh committed:
create a new folder controller 

moved functionality of user's (aka auth) api routes as well as category for better organization

Kanchanakkh commented:
Thank you for the feedback. I will try my best to fix them.

dang8577 commented:
There are some bugs such as when going back to another page, the link doesn't change, I still can't register. Code still is unstable
